AdobeStockAsset/Model/AppendAttributes.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\AdobeStockAsset\Model;
+
+use Magento\AdobeStockAssetApi\Api\Data\AssetInterface;
+use Magento\Framework\Api\AttributeValueFactory;
+use Magento\Framework\Api\Search\SearchResultInterface;
+use Magento\Framework\App\ResourceConnection;
+
+/**
+ * Class AddIsDownloadedToSearchResult
+ */
+class AppendAttributes
+{
+    const TABLE_ASSET                  = 'adobe_stock_asset';
+    const ATTRIBUTE_CODE_IS_DOWNLOADED = 'is_downloaded';
+    const ATTRIBUTE_CODE_PATH          = 'path';
+
+    /**
+     * @var AttributeValueFactory
+     */
+    private $attributeValueFactory;
+
+    /**
+     * @var ResourceConnection
+     */
+    private $resourceConnection;
+
+    /**
+     * AddIsDownloadedToSearchResult constructor.
+     * @param ResourceConnection $resourceConnection
+     * @param AttributeValueFactory $attributeValueFactory
+     */
+    public function __construct(
+        ResourceConnection $resourceConnection,
+        AttributeValueFactory $attributeValueFactory
+    ) {
+        $this->resourceConnection = $resourceConnection;
+        $this->attributeValueFactory = $attributeValueFactory;
+    }
+
+    /**
+     * Add additional asset attributes
+     *
+     * @param SearchResultInterface $searchResult
+     * @return SearchResultInterface
+     */
+    public function execute(SearchResultInterface $searchResult): SearchResultInterface
+    {
+        $items = $searchResult->getItems();
+        $itemIds = [];
+
+        foreach ($items as $key => $item) {
+            $itemIds[$key] = $item->getId();
+        }
+
+        if (count($itemIds)) {
+            $connection = $this->resourceConnection->getConnection();
```

Comment:
Can you please move this select from `adobe_stock_asset` table to a separate class to be able to reuse that, and place this class in the `ResourceModel` namespace in order to indicate the access to the database. I'd suggest moving to `\Magento\AdobeStockAsset\Model\ResourceModel\Asset\Load::execute(array $ids): array`

Reply:
It would also be great if you could cover these files with unit tests